### PR TITLE
Add support for request helper

### PR DESCRIPTION
**Pull request by j3j5**
Add support for the `request()` helper.

This is my first significant contribution to phpstan/larastan so all feedback is welcome.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Looks very nice. Does it work?
BTW Larastan's code is checked by PHPStan ;-)

**Comment by szepeviktor** [comment]
Thank you for your contribution!

File: src/ReturnTypes/Helpers/RequestExtension.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        FuncCall $functionCall,
+        Scope $scope
+    ): Type {
+        // No arguments, it returns a request object from the container
+        if (count($functionCall->args) === 0) {
+            return new ObjectType(\Illuminate\Http\Request::class);
+        }
+
+        if (isset($functionCall->args[0]->value) && $functionCall->args[0]->value instanceof Array_) {
+            return new ArrayType(new MixedType(), new MixedType());
```
Comment (nunomaduro) [inline src/ReturnTypes/Helpers/RequestExtension.php]: Shouldn't this be only `return new ArrayType()` ?

File: src/ReturnTypes/Helpers/RequestExtension.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Larastan.
+ *
+ * (c) Julio Foulquie <jfoulquie@gmail.com>
```
Comment (nunomaduro) [inline src/ReturnTypes/Helpers/RequestExtension.php]: Sorry, but this should contain the name of the package owner: `Nuno Maduro <enunomaduro@gmail.com>`.

Comment (j3j5) [reply src/ReturnTypes/Helpers/RequestExtension.php]: I'll change it! I wasn't sure what was the licensing policy for fully new files.

Comment (nunomaduro) [reply src/ReturnTypes/Helpers/RequestExtension.php]: @j3j5 Thanks!

File: src/ReturnTypes/Helpers/RequestExtension.php (hunk 1 above)
Comment (j3j5) [reply src/ReturnTypes/Helpers/RequestExtension.php]: Looking at the constructor I didn't think it was possible, but I just tested it and it works. What are the values of `Type $keyType` and `Type $itemType` when you construct the object w/o arguments?

**Comment by j3j5** [comment]
I've pushed to the original branch, not sure how GH will handle now that the PR is already merged. Shall I submit another one?

**Comment by nunomaduro** [comment]
@j3j5 I think you can submit a new PR.

File: src/ReturnTypes/Helpers/RequestExtension.php (hunk 1 above)
Comment (nunomaduro) [reply src/ReturnTypes/Helpers/RequestExtension.php]: When $key is an `array` the request helper returns the result of the method `only`:
```
        if (is_array($key)) {
            return app('request')->only($key);
        }
```

And the only method returns an `array`:
```php
    /**
     * Get a subset containing the provided keys with values from the input data.
     *
     * @param  array|mixed  $keys
     * @return array
     */
    public function only($keys)
```

Comment (j3j5) [reply src/ReturnTypes/Helpers/RequestExtension.php]: Yes, that I understand, that's why I returned a `PHPStan\Type\ArrayType` in the first place. But looking at the source code of that class, I interpreted (wrongly it seems) that the constructor needed 2 arguments.

```php
public function __construct(Type $keyType, Type $itemType)
{
	if ($keyType->describe(VerbosityLevel::value()) === '(int|string)') {
		$keyType = new MixedType();
	}
	$this->keyType = $keyType;
	$this->itemType = $itemType;
}
```
I looked at PHPStan's source code and I copypasted that one, thinking it made sense.

Comment (j3j5) [reply src/ReturnTypes/Helpers/RequestExtension.php]: LOL, this is what I was talking about https://travis-ci.org/nunomaduro/larastan/jobs/433160174#L650-L659

**Pull request merged**